test/unit/calculators/plan_adoption_leave_test.rb

```diff
@@ -69,6 +60,30 @@ class PlanAdoptionLeaveTest < ActiveSupport::TestCase
           end
         end
       end
+
+      context "test validators" do
+        should "arrival date before match date should be invalid" do
+          @calculator.arrival_date = Date.parse("2012-05-25")
+          assert_not @calculator.valid_arrival_date?
+        end
```

Comment:
If the `arrival_date` and `start_date` were wrapped in methods, I think these tests could be made more readable.

Also the dates could be randomised or incremented with time to avoid the possibility that dates like "2012-12-25" are edge or special cases.

For example:
```ruby
def start_date
  @start_date ||= 1.month.ago
end

def arrival_date
  start_date + 14.days
end

def setup
  @calculator.start_date = start_date
  @calculator.arrival_date = arrival_date
end

should "arrival date before match date should be invalid" do
  @calculator.arrival_date = start_date - 1.day
  assert_not @calculator.valid_arrival_date?
end
```